src/figuresClasses.ts

```diff
@@ -1,19 +1,95 @@
-export interface Figure {
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle'
+}
 
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue'
 }
 
-export class Triangle {
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if ([a, b, c].some((side) => side <= 0)) {
```

Comment:
Better create a helper function that will accept arguments (any length) and determine if some have `value <= 0`.
```
(...args) => args.some() ....
```
And you could use it for any figure you need to check. Now almost the same check you have in 3 places all over the code.

Reply:
Yes, this is a really cool improvement